cpp/src/fil/internal.cuh

```diff
@@ -266,6 +301,145 @@ struct forest_params_t {
 /// FIL_TPB is the number of threads per block to use with FIL kernels
 const int FIL_TPB = 256;
 
+const uint32_t MAX_PRECISE_INT_FLOAT = 1 << 24;  // 16'777'216
+
+__host__ __device__ __forceinline__ int fetch_bit(const uint8_t* array, int bit)
+{
+  return (array[bit / BITS_PER_BYTE] >> (bit % BITS_PER_BYTE)) & 1;
+}
+
+struct cat_feature_counters {
+  int max_matching = -1, n_nodes = 0;
+};
+
+struct categorical_sets {
+  // arrays from each node ID are concatenated first, then from all categories
+  const uint8_t* bits = nullptr;
```

Comment:
We probably want to use `std::uint8_t` throughout and `#include <cstdint>`. If we must use `uint8_t` directly, we should at least `#include <stdint.h>`

Reply:
are we introducing a new standard to include all headers directly even if they're already included indirectly?
from a glance at rapids codebase, this is not the standard so far